File: 05.RestSharp-and-RestAssured/02-petstore-rest-assured/src/main/java/helpers/RandomDataGenerator.java
package helpers;

import java.util.Random;

public class RandomDataGenerator {
    private static final Random random = new Random();

    public static Long generateId() {
        return 100000L + random.nextInt(900000);
    }

    public static String generateUsername() {
        return "user_" + System.currentTimeMillis();
    }

    public static String generateEmail() {
        return "email_" + System.currentTimeMillis() + "@example.com";
    }

    public static String generatePhone() {
        return "1" + (100000000L + random.nextInt(900000000));
    }
}

File: 01.Playwright/03-automationexercise-ui-csharp/AutomationExercise.Tests/Helpers/RandomDataGenerator.cs
using System;

namespace AutomationExercise.Tests.Helpers
{
    public static class RandomDataGenerator
    {
        public static string GenerateEmail(string prefix = "qa_user")
        {
            var unique = Guid.NewGuid().ToString("N").Substring(0, 10);
            return $"{prefix}_{unique}_{Random.Shared.Next(1000, 9999)}@example.com";
        }

        public static string GenerateName(string prefix = "User")
        {
            var unique = Guid.NewGuid().ToString("N").Substring(0, 8);
            return $"{prefix}_{unique}_{Random.Shared.Next(1000, 9999)}";
        }
    }
}
